CONTRIBUTING.md

```diff
@@ -44,6 +44,28 @@ for example:
 git clone git@github.com:YOUR-USERNAME/narwhals.git
 ```
 
+### 3. Add the `upstream` remote
```

Comment:
Should all the numbers be updated? Or this could be a sub-section within 3 ✨

Reply:
Updated. thank you.